core-ui/src/shared/components/ResourceRef/SecretRef.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { useGetList } from 'react-shared';
+
+import { ExternalResourceRef } from './ExternalResourceRef';
+
+const SECRETS_URL = '/api/v1/secrets';
+
+export function SecretRef({ labelSelector, ...props }) {
+  const url = labelSelector
+    ? `${SECRETS_URL}?labelSelector=${labelSelector}`
```

Comment:
Should we show all secrets? If I create with any opaque secret, I get ```Secret is not if type kubernetes.io/tls"```. Maybe we should only show those? (But then we should add a tooltip saying that only those are valid secrets in this scenario)